test/test_model.py

```diff
@@ -207,6 +207,52 @@ def test_model_info(self):
         self.assertIn('theta', model_info_include['parameters'])
         self.assertIn('included_files', model_info_include)
 
+    def test_compile_with_includes(self):
+        getmtime = os.path.getmtime
+        configs = [
+            ('add_one_model.stan', ['include-path']),
+            ('bernoulli_include.stan', []),
+        ]
+        for stan_file, include_paths in configs:
+            stan_file = os.path.join(DATAFILES_PATH, stan_file)
+            include_paths = [
+                os.path.join(DATAFILES_PATH, path) for path in include_paths
+            ]
+
+            # Compile for the first time.
+            model = CmdStanModel(
+                stan_file=stan_file,
+                compile=False,
+                stanc_options={"include-paths": include_paths},
+            )
+            with LogCapture(level=logging.INFO) as log:
+                model.compile()
+            log.check_present(
+                ('cmdstanpy', 'INFO', StringComparison('compiling stan file'))
+            )
+
+            # Compile for the second time, ensuring cache is used.
+            with LogCapture(level=logging.DEBUG) as log:
+                model.compile()
+            log.check_present(
+                ('cmdstanpy', 'DEBUG', StringComparison('found newer exe file'))
+            )
+
+            # Compile after modifying included file, ensuring cache is not used.
+            def _patched_getmtime(filename: str) -> float:
+                includes = ['divide_real_by_two.stan', 'add_one_function.stan']
+                if any(filename.endswith(include) for include in includes):
```

Comment:
Could this be more general but just saying that the filename does _not_ end with the `stan_file` parameter?